pkg/harness/stackrun/helpers.go

```diff
@@ -34,8 +37,63 @@ func StartStackRun(client console.Client, id string) error {
 	return MarkStackRun(client, id, gqlclient.StackStatusRunning)
 }
 
-func CompleteStackRun(client console.Client, id string) error {
-	return MarkStackRun(client, id, gqlclient.StackStatusSuccessful)
+func CompleteStackRun(client console.Client, id string, attributes *gqlclient.StackRunAttributes) (err error) {
+	if attributes == nil {
+		return fmt.Errorf("cannot complete stack run with nil attributes")
+	}
+
+	createModifierIter := func() func() func(attributes *gqlclient.StackRunAttributes) *gqlclient.StackRunAttributes {
+		idx := 0
+		modifiers := []func(attributes *gqlclient.StackRunAttributes) *gqlclient.StackRunAttributes{
+			func(attributes *gqlclient.StackRunAttributes) *gqlclient.StackRunAttributes {
+				if attributes.State == nil {
+					return attributes
+				}
+
+				// first drop the state
+				attributes.State.State = nil
+				return attributes
+			},
+			func(attributes *gqlclient.StackRunAttributes) *gqlclient.StackRunAttributes {
+				if attributes.State == nil {
+					return attributes
+				}
+
+				// next drop the plan
+				attributes.State.Plan = nil
+				return attributes
+			},
+			func(attributes *gqlclient.StackRunAttributes) *gqlclient.StackRunAttributes {
+				// as the last resort drop output
+				attributes.Output = nil
```

Comment:
we shouldn't ever drop this (although in theory it could cause an issue). The reason is outputs could be piped to services and that could disrupt other things in the system

Reply:
sure, removed